helm-charts/splunk-otel-collector/templates/operator/_helpers.tpl

```diff
@@ -163,6 +166,11 @@ Helper for generating environment variables for each instrumentation library.
 
   {{- /* Output final OTEL_EXPORTER_OTLP_ENDPOINT, if applicable based on input conditions */ -}}
   {{- if $customOtelExporterEndpoint }}
+    {{- /* {{- /* Ensure the SPLUNK_OTEL_AGENT env var is set with per language env vars to successfully use it in env var substitution */ -}}
+    {{- if contains "SPLUNK_OTEL_AGENT" $customOtelExporterEndpoint -}}
+      {{- printf "- name: SPLUNK_OTEL_AGENT\n  valueFrom:\n    fieldRef:\n      apiVersion: v1\n      fieldPath: status.hostIP" | indent 6 }}
```

Comment:
Small feedback - the readability of these helpers would be better if we could keep yaml snippets as-is. But I understand that would require additional changes to how this is organized. Maybe something we can slowly improve upon.

Reply:
I agree that readability could be improved. Initially, we aimed to use multiple helm template files/snippets, but the complexity escalated quickly for our needs, pushing the limits of what templates could handle. This design choice was driven by the need for seamless adoption of new operator versions, automatically integrating new upstream instrumentation languages through automation, which necessitated dynamic values and Helm Helpers.

Adding unit tests to showcase the creation and management of the Operator Instrumentation config would indeed be beneficial.